### PR TITLE
Amit/roger 989 Added authorize logic for allowed users and actions

### PR DESCRIPTION
**Pull request by amitbose327**
@vmahedia , @ankanm and @04msambit Please review:

### CODE REVIEW

**Comment by 04msambit** [comment]
LGTM

File: aaad/authorizers.py [hunk 1]
```diff
@@ -7,21 +8,62 @@ def __init__(self, filename):
...
+
+            for u in data[user]['can_act_as']:
+                self.get_merged_data(u, allowed_users, allowed_actions, data, action)
+
+        def resource_check(self, resource, allowed_actions):
+            for pattern in allowed_actions:
+                prog = re.compile(pattern)
+                result = prog.match(resource)
+                if result:
+                    return True
```
Comment (ankanm) [inline aaad/authorizers.py]: End `resource_check` with `return False.

File: ansible/roles/generate-permission-files/templates/user-permissions.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,47 @@
+admin:
```
Comment (ankanm) [inline ansible/roles/generate-permission-files/templates/user-permissions.yml.j2]: This should probably be better in the files section as it's not a template.

File: ansible/roles/generate-permission-files/tasks/main.yml [hunk 3]
```diff
@@ -163,7 +163,7 @@
     when: enable_nginx_auth_proxy is defined and enable_nginx_auth_proxy|bool and
           nginx_auth_permissions_local_path is not defined and
           auth_permissions_gspreadsheets is not defined
-    template: src=user-permissions.yml.j2 dest="{{nginx_conf_dir}}/user-permissions.yml" owner=root group=root mode=0644
+    template: src={{ role_path }}/files/user-permissions.yml dest="{{nginx_conf_dir}}/user-permissions.yml" owner=root group=root mode=0644
```
Comment (ankanm) [inline ansible/roles/generate-permission-files/tasks/main.yml]: not `template` anymore

File: ansible/roles/generate-permission-files/tasks/main.yml [hunk 4]
```diff
@@ -163,7 +163,7 @@
     when: enable_nginx_auth_proxy is defined and enable_nginx_auth_proxy|bool and
           nginx_auth_permissions_local_path is not defined and
           auth_permissions_gspreadsheets is not defined
-    template: src=user-permissions.json.j2 dest="{{nginx_conf_dir}}/user-permissions.json" owner=root group=root mode=0644
+    template: src={{ role_path }}/files/user-permissions.yml dest="{{nginx_conf_dir}}/user-permissions.yml" owner=root group=root mode=0644
```
Comment (ankanm) [inline ansible/roles/generate-permission-files/tasks/main.yml]: `copy`

**Comment by ankanm** [comment]
LG apart from comment above.

File: ansible/roles/generate-permission-files/tasks/main.yml (hunk 4 above)
Comment (amitbose327) [inline ansible/roles/generate-permission-files/tasks/main.yml]: Done

**Pull request merged**